### ISSUE TITLE
Pod migrate to another node that replica will lost tables by using local pv 

### ISSUE THREAD

**Issue by godliness** [Altinity/clickhouse-operator#776]
When node under the system resource pressure, kubelet will evict pod in the node, clickhouse-server will migrate to another node, the new migrated replica will lost tables.

Overall, if we using local storage, replica will lost all tables so that it can not sync data from other alive replicas.

I think we did nothing when new pod added, is that right?

**Comment by Slach** [comment]
did you use `volumeClaimTemplates` with properly storage class which provisioned independently from the node and have CSIDriver which allow move persistent volumes between nodes in your `kind: ClickhouseInstallation` ?

**Comment by godliness** [comment]
> did you use `volumeClaimTemplates` with properly storage class which provisioned independently from the node and have CSIDriver which allow move persistent volumes between nodes in your `kind: ClickhouseInstallation` ?

we didn't use storage class which be implemented by csi plugin，s3，ceph，and so on. because local storage could provide best performance for database.

**Comment by Slach** [comment]
localPath provisioner doesn't allow move PV between nodes, so data lose is expected

`clickhouse-operator` itself doesn't manage disk storages, it use only Persistent Volume Claims which implemented on Kubernetes side and depends on your current cluster CSI drivers

**Comment by godliness** [comment]
> localPath provisioner doesn't allow move PV between nodes, so data lose is expected
> 
> `clickhouse-operator` itself doesn't manage disk storages, it use only Persistent Volume Claims which implemented on Kubernetes side and depends on your current cluster CSI drivers

Thanks, @Slach
Last question:
Replicas in the same shard migrate to another node didn’t do migrateTables when use localpath provisioner, is that also expected?

**Comment by Slach** [comment]
yes, clickhouse-operator doesn't directly resposible to re-schedule pod between node, it only reconcile `kind: ClickhouseInstallation`  custom resources, and render `kind: StatefulSet` according to `spec.configuration.clusters` and can only define podAntiAffinity or podAffinity from `spec.templates.podTemplates`

`clickhouse-operator` will try to replicate schema to new clickhouse-server instance
when you change `spec.clusters[].layout`

**Comment by godliness** [comment]
> yes, clickhouse-operator doesn't directly resposible to re-schedule pod between node, it only reconcile `kind: ClickhouseInstallation` custom resources, and render `kind: StatefulSet` according to `spec.configuration.clusters` and can only define podAntiAffinity or podAffinity from `spec.templates.podTemplates`
> 
> `clickhouse-operator` will try to replicate schema to new clickhouse-server instance
> when you change `spec.clusters[].layout`

Thanks! I understood.

Local PV has no relationship with pod migrate between nodes, if i use local storage, pod must affinity with this node. So i must keep multi replicas for shard to support high available.

**Comment by Slach** [comment]
@godliness yes, you absolutely right
